contracts/src/finance/vesting_wallet.rs

```diff
@@ -0,0 +1,617 @@
+    /// [`Error::InvalidToken`] is returned.
+    ///
+    /// # Panics
+    ///
+    /// If `total_allocation * (timestamp - self.start())` exceeds `U256::MAX`.
+    #[selector(name = "releasable")]
+    fn releasable_erc20(&self, token: Address) -> Result<U256, Self::Error>;
+
+    /// Release the native token (ether) that have already vested.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    ///
+    /// # Errors
+    ///
+    /// If ETH transfer fails, then the error [`Error::ReleaseEtherFailed`] is
+    /// returned.
+    ///
+    /// # Events
+    ///
+    /// Emits an [`EtherReleased`] event.
+    ///
+    /// # Panics
+    ///
+    /// If total allocation exceeds `U256::MAX`.
+    /// If `total_allocation * (timestamp - self.start())` exceeds `U256::MAX`.
+    #[selector(name = "release")]
+    fn release_eth(&mut self) -> Result<(), Self::Error>;
+
+    /// Release the tokens that have already vested.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `token` - Address of the token being released.
+    ///
+    /// # Errors
+    ///
+    /// If the `token` address is not a contract, then the error
+    /// [`Error::InvalidToken`] is returned.
+    /// If the contract fails to execute the call, then the error
+    /// [`Error::ReleaseTokenFailed`] is returned.
+    ///
+    /// # Events
+    ///
+    /// Emits an [`ERC20Released`] event.
+    ///
+    /// # Panics
+    ///
+    /// If `total_allocation * (timestamp - self.start())` exceeds `U256::MAX`.
+    #[selector(name = "release")]
+    fn release_erc20(&mut self, token: Address) -> Result<(), Self::Error>;
+
+    /// Calculates the amount of ether that has already vested. Default
+    /// implementation is a linear vesting curve.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `timestamp` - Point in time for which to check the vested amount.
+    ///
+    /// # Panics
+    ///
+    /// If total allocation exceeds `U256::MAX`.
+    /// If `total_allocation * (timestamp - self.start())` exceeds `U256::MAX`.
+    #[selector(name = "vestedAmount")]
+    fn vested_amount_eth(&self, timestamp: u64) -> U256;
+
+    /// Calculates the amount of tokens that has already vested. Default
+    /// implementation is a linear vesting curve.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token` - Address of the token being released.
+    /// * `timestamp` - Point in time for which to check the vested amount.
+    ///
+    /// # Errors
+    ///
+    /// If the `token` address is not a contract, then the error
+    /// [`Error::InvalidToken`] is returned.
+    ///
+    /// # Panics
+    ///
+    /// If `total_allocation * (timestamp - self.start())` exceeds `U256::MAX`.
+    #[selector(name = "vestedAmount")]
```

Comment:
I believe `#[selector(name = "...")]` is not needed in trait definition.

Reply:
Can it be removed from [IErc721::safe_transfer_from_with_data](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/token/erc721/mod.rs#L329) too?